config/routes.rb

```diff
@@ -8,5 +8,9 @@
   get "/users/:id", to: "users#show"
   
   get "/users/:id/posts", to: "posts#index"
+  get "/users/:id/posts/new", to: "posts#new"
+  post "/users/:id/posts/create", to: "posts#create"
   get "/users/:id/posts/:id_post", to: "posts#show"
+  post "/users/:id/posts/:id_post/add_comment", to: "posts#add_comment", as: "add_comment"
+  post "/users/:id/posts/:id_post/add_like", to: "posts#add_like", as: "add_like"
 end
```

Comment:
- [ ] Once you have update all the above changes, you should make sure you add all the appropriate `routes` such as for `creating posts`, `creating comments`, `creating likes`,...
I can suggest you just summarize all your routes to this example below:
```ruby

 root "users#index"

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:new, :create]
    end
  end
```